src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -3347,6 +3403,38 @@ private static bool CheckValEscape(ImmutableArray<BoundExpression> expressions,
             return true;
         }
 
+        private static bool CheckInterpolatedStringExpressionEscape(BoundInterpolatedString interpolatedString, uint escapeFrom, uint escapeTo, BindingDiagnosticBag diagnostics, bool isRef)
+        {
+            // We need to check to see if any values could potentially escape outside the max depth via the handler type.
+            // Consider the case where a ref-struct handler saves off the result of one call to AppendFormatted,
+            // and then on a subsequent call it either assigns that saved value to another ref struct with a larger
+            // escape, or does the opposite. In either case, we need to check.
+
+            foreach (var part in interpolatedString.Parts)
+            {
+                if (part is not BoundCall { Method: { Name: "AppendFormatted" } } call)
+                {
+                    // Dynamic calls cannot have ref struct parameters, and AppendLiteral calls will always have literal
+                    // string arguments and do not require us to be concerned with escape
+                    continue;
+                }
+
+                // The interpolation component is always the first argument to the method, and it was not passed by name
+                // so there can be no reordering.
+                var argument = call.Arguments[0];
+                var success = isRef
```

Comment:
For expressions passed via the holes, we check `GetRefEscape` or `GetValEscape` depending on , but for arguments passed via argument attribute we only check the val-escape (`ValSafeToEscape`). Just want to confirm that's intentional/correct.

Reply:
Mm... it looks like this method is only ever called with `isRef` `false`.

Reply:
I'm also missing a check on the constructor, I'll add that as well.